backend/test_context_selection.py

```
@@ -0,0 +1,163 @@
+        
+        print(f"📝 Relevant Memories ({len(relevant_memories)}):")
+        for memory in relevant_memories:
+            print(f"  • [{memory.type.upper()}] {memory.content}")
+        
+        print(f"📋 Relevant Tasks ({len(relevant_tasks)}):")
+        for task in relevant_tasks:
+            status_emoji = {"pending": "⏳", "in_progress": "🔄", "completed": "✅"}.get(task.status, "📋")
+            print(f"  • {status_emoji} {task.title} - {task.description}")
+    
+    # Test hierarchical context
+    print(f"\n🏗️  Testing Hierarchical Context Selection")
```

Ruff (0.12.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)